pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java

```diff
@@ -81,10 +81,11 @@ public void onManagedLedgerPropertiesInitialize(Map<String, String> propertiesMa
 
     @Override
     public void onManagedLedgerLastLedgerInitialize(String name, LedgerHandle lh) {
+        LedgerEntries ledgerEntries = null;
         try {
             for (BrokerEntryMetadataInterceptor interceptor : brokerEntryMetadataInterceptors) {
                 if (interceptor instanceof AppendIndexMetadataInterceptor && lh.getLastAddConfirmed() >= 0) {
-                    LedgerEntries ledgerEntries =
+                    ledgerEntries =
```

Comment:
What happens if we get to here multiple times?
There is no invariant that enforces the presence of only one AppendIndexMetadataInterceptor

Reply:
You can use a try-with-resources block and close the object as soon as we do not it anymore

Reply:
There will be only on `AppendIndexMetadataInterceptor` for each `ManagedLedger`

Reply:
what about
```
try (LedgerEntries ledgerEntries = lh.read....) {
...
}
```

Reply:
try-with-resources is alternative way and makes code more tidy. 
 I think it's better to raise an individual issue for refactor code like try-with-resources